src/transports/http.rs

```diff
@@ -1,283 +1,176 @@
-                ResponseState::Waiting(ref mut waiting) => {
-                    log::trace!("[{}] Checking response.", id);
-                    let response = ready!(waiting.poll_unpin(ctx))?;
-                    if !response.status().is_success() {
-                        return Poll::Ready(Err(Error::Transport(format!(
-                            "Unexpected response status code: {}",
-                            response.status()
-                        ))));
-                    }
-                    self.state = ResponseState::Reading(Box::pin(response.bytes()));
-                }
-                ResponseState::Reading(ref mut body) => {
-                    log::trace!("[{}] Reading body.", id);
-                    let chunk = ready!(body.poll_unpin(ctx))?;
-                    let response = chunk.to_vec();
-                    log::trace!(
-                        "[{}] Extracting result from:\n{}",
-                        self.id,
-                        std::str::from_utf8(&response).unwrap_or("<invalid utf8>")
-                    );
-                    return Poll::Ready((self.extract)(response));
-                }
-            }
-        }
+fn id_of_output(output: &Output) -> Result<RequestId> {
+    let id = match output {
+        Output::Success(success) => &success.id,
+        Output::Failure(failure) => &failure.id,
+    };
+    match id {
+        jsonrpc_core::Id::Num(num) => Ok(*num as RequestId),
+        _ => Err(Error::InvalidResponse("response id is not u64".to_string())),
     }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
 
-    #[test]
-    fn http_supports_basic_auth_with_user_and_password() {
-        let http = Http::new("https://user:password@127.0.0.1:8545").unwrap();
-        assert!(http.basic_auth.is_some());
-        assert_eq!(
-            http.basic_auth,
-            Some(HeaderValue::from_static("Basic dXNlcjpwYXNzd29yZA=="))
-        )
-    }
-
-    #[test]
-    fn http_supports_basic_auth_with_user_no_password() {
-        let http = Http::new("https://username:@127.0.0.1:8545").unwrap();
-        assert!(http.basic_auth.is_some());
-        assert_eq!(http.basic_auth, Some(HeaderValue::from_static("Basic dXNlcm5hbWU6")))
-    }
-
-    #[test]
-    fn http_supports_basic_auth_with_only_password() {
-        let http = Http::new("https://:password@127.0.0.1:8545").unwrap();
-        assert!(http.basic_auth.is_some());
-        assert_eq!(http.basic_auth, Some(HeaderValue::from_static("Basic OnBhc3N3b3Jk")))
-    }
-
     async fn server(req: hyper::Request<hyper::Body>) -> hyper::Result<hyper::Response<hyper::Body>> {
         use hyper::body::HttpBody;
 
-        let expected = r#"{"jsonrpc":"2.0","method":"eth_getAccounts","params":[],"id":1}"#;
-        let response = r#"{"jsonrpc":"2.0","id":1,"result":"x"}"#;
+        let expected = r#"{"jsonrpc":"2.0","method":"eth_getAccounts","params":[],"id":0}"#;
+        let response = r#"{"jsonrpc":"2.0","id":0,"result":"x"}"#;
```

Comment:
There is an old commit from several years ago that changed the initial id value from 0 to 1 in Http::new. I don't see why so I changed it back to 0.

Reply:
Well, it's only 12 months old, not several years :) I don't think there was any concrete reason to change that apart from aesthetics.